### PR TITLE
core-app-api,core-plugin-api: remove MUI dependencies

### LINKED ISSUE #7692: core-plugin-api: Remove dependency on theme
We're unable to depend on `theme` in core-plugin-api let's make sure it's removed to be able to ship version 1 of the core packages.

Make sure `theme` is removed from
https://github.com/backstage/backstage/blob/master/packages/core-plugin-api/package.json

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

More core cleanup, these deprecations are complete so just need to remove the deps 😁 

Fixes #7692

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by Rugvip** [comment]
Needs #8737 to be released first. We'll likely release this in the release just after (1 ½ weeks later) since `OldIconComponent` was never intended to be exported and likely has very little existing usage.

**Review by freben — APPROVED** [review]

**Comment by freben** [comment]
Bunch of conflicts now of course.

**Review by benjdlambert — APPROVED** [review]

File: .changeset/twelve-bears-fail.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/core-app-api': patch
+'@backstage/core-plugin-api': patch
```
Comment (benjdlambert) [inline .changeset/twelve-bears-fail.md:3]: Shouldn't this be `minor` as it's now been removed?

Comment (Rugvip) [reply .changeset/twelve-bears-fail.md:3]: 👍 , added a separate changeset for the removal

**Review by jhaals — APPROVED** [review]

**Comment by benjdlambert** [comment]
Sweet - happy to merge this as soon as it's ready so we don't get any other conflicts 👍

**Pull request merged**